Update some of our more 'dated' instructions

milljm opened the pull request:
Update the version of PETSc we were asking users to install.
Update the Windows 10 instructions. As well as alert the user as to its current supportability.

Closes #11724

Commit:
Update some of our more 'dated' instructions

Update the version of PETSc we were asking users to install.
Update the Windows 10 instructions. As well as alert the user as to its current supportability.

Closes #11724

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/203347/) on 18f6151 wanted to post the following:

View the site [here](http://mooseframework.inl.gov/docs/PRs/11727/site)

This comment will be updated on new commits.

aeslaughter commented:
@milljm It would be simple to add an extension to use a file for the PETSc version, then in the markdown you could do something like:

```markdown
This is really cool, the current version of PETSc is !!package petsc!!, ...
```

I can help you get started, the "python/MooseDocs/extensions/acroynm.py" does something similar.

http://www.mooseframework.org/moose/utilities/MooseDocs/extensions/acronym.html#bef8d33b-4513-4f57-b8e8-ddeac08fcccd

aeslaughter commented:
You can add this feature to the "package.py" extension.

aeslaughter commented:
Another thing, while you are in here, the "getting started" instructions should be moved into modules/combined/doc/content... since they are specific for our website. That is, other applications (BISON, MASTODON, etc.) have their own set of instructions.

This will also allow us to remove the "packages" extension being available by default, since again this is really a MOOSE specific thing.

Commit:
Move 'getting started', use version templates

Move the getting started packages to a more moose-only related location.
Create ability to use the MooseDoc's extension system to replace all version
assignments with those in a yaml template file.

aeslaughter reviewed (changes requested):
Please add comments to these classes and methods

Commit:
Fix pyflakes issues, templating syntax, grammar...

Commit:
Adding docstrings

Commit:
Move the getting started pages back into framework

Commit:
Refactor the cloning and building instructions

Cloning and building MOOSE is straight forward enough to belong in its own
markdown file to be sourced by other documents.

Moved the Minimum Requires document to be rendered at the top of each set
of instructions.

jwpeterson commented on framework/doc/content/getting_started/installation/cluster.md:
````diff
@@ -1,178 +1,205 @@
+module avail
+```
 
+Verify that somewhere among all the other modules installed, you see: 'moose-dev-gcc' among them.
````
typo: remove "among them" from end of sentence.

jwpeterson commented on framework/doc/content/getting_started/installation/windows10.md:
```diff
@@ -1,42 +1,54 @@
+  - (at the time of this writing, we have installation packages for: Ubuntu and OpenSUSE Leap)
+!alert-end!
 
+Begin by performing the following external sets of instructions. Do note, that when it comes time to deciding what distribution of Linux you want, you should try and choose a flavor listed among our getting started pages for the smoothest installation process. This is not a requirement however, as MOOSE will run on just about any flavor of Linux if you are of the adventurous type!
```
typo: "comes time to deciding" -> "comes time to decide"

jwpeterson reviewed (commented):
@milljm I just found a few grammatical things I'd like to fix. If it's OK with you I will just push a commit directly to your branch that fixes them up.

jwpeterson committed:
Fix minor grammatical things.

Commit:
use templates for minimum_requirements

moved the packages_config version file to its proper location

Commit:
unify cluster instructions with manual gcc/llvm

Add miniconda/peacock instructions to the manual_gcc and llvm instructions.

modify cluster/gcc/llvm instructions to work with a single miniconda, petsc template.
